media/src/project/project-map-pane-panels.tsx

```diff
@@ -464,6 +465,154 @@ export const EventDetailPanel: React.FC<EventDetailPanelProps> = (
     );
 };
 
+interface ActivityProps {
+    activity: ActivityData | null;
+    createActivity(instructions: string): void;
+    updateActivity(instructions: string, pk: number): void;
+    deleteActivity(pk: number): void;
+}
+
+const Activity: React.FC<ActivityProps> = (
+    {activity, createActivity, updateActivity, deleteActivity}: ActivityProps) => {
+
+    const [instructions, setInstructions] = useState<string | null>(null);
+    const [showMenu, setShowMenu] = useState<boolean>(false);
+    const [showCreateForm, setShowCreateForm] = useState<boolean>(false);
+    const [showEditForm, setShowEditForm] = useState<boolean>(false);
+
+    const handleCreateActivity = (e: React.FormEvent<HTMLFormElement>): void => {
+        e.preventDefault();
+        instructions && createActivity(instructions);
```

Comment:
I'm somewhat uncomfortable about this pattern, and see that it is repeated below. Maybe I'm just not familiar with it? Let's chat.